libraries/Microsoft.Bot.Builder.Azure/AzureBlobStorage.cs

```diff
@@ -108,8 +108,18 @@ public AzureBlobStorage(CloudStorageAccount storageAccount, string containerName
 
             await Task.WhenAll(readTasks);
 
-            // Project back the entries that were read, filtering out any entries that were not found
-            return readTasks.Select(readTask => readTask.Result).Where(kvp => kvp.Key != null);
+            // Project back the entries that were read, filtering out any entries that were not found.
+            // This gives us a List<KeyValuePair<>>. 
+            var items = readTasks.Select(readTask => readTask.Result).Where(kvp => kvp.Key != null);
+
+            // Iterate over the KVPs and add them into a dictionary. 
+            IDictionary<string, object> values = new Dictionary<string, object>();

```

Comment:
Suggest you "right size" the dictionary's capacity up front based on the number of keys to avoid any growth cycles:

```CSharp
var values = new Dictionary<string, object>(keys.Length);
```

Reply:
Same comment. This declaration is gone now in favor of the LINQ .ToDictionary().